packages/tasks/src/model-libraries-snippets.ts

```diff
@@ -1498,7 +1498,7 @@ export const transformers = (model: ModelData): string[] => {
 			`${processorVarName} = ${info.processor}.from_pretrained("${model.id}"` + remote_code_snippet + ")",
 			`model = ${info.auto_model}.from_pretrained("${model.id}"` + remote_code_snippet + ")"
 		);
-		if (model.tags.includes("conversational")) {
+		if (model.tags.includes("conversational") && model.config?.tokenizer_config?.chat_template) {
```

Comment:
I'm not sure if this is populated if i.e. `chat_template.jinja` exists but it's not in `tokenizer_config.json`

Reply:
(I think it's not)

Reply:
The correct way to check if a chat template is set is:

```js
			(this.config?.tokenizer_config?.chat_template !== undefined ||
				this.config?.processor_config?.chat_template !== undefined ||
				this.config?.chat_template_jinja !== undefined
```

(taken from [internal repo](https://github.com/huggingface-internal/moon-landing/blob/3ca0960c3e572aed8f92b588a6f707933c9a46ce/server/repo_types/ModelInfo.ts#L700))

Note that `this.config?.chat_template_jinja` is the most future-proof field (but is not populated for all models with a chat template)

Reply:
agree^